src/js/background.js

```diff
@@ -824,8 +825,9 @@ Badger.prototype = {
    * Merge data exported from a different badger into this badger's storage.
    *
    * @param {Object} data the user data to merge in
+   * @param {Undefined} runInMigration undefined in all cases except when this function is run in a migration, then prevents infinite loop
```

Comment:
Should be something like the following (JSDoc convention for an optional boolean that defaults to false, also naming convention: let's use Java naming for objects/object properties/methods, and C naming for primitives):
```
@param {Boolean} [from_migration=false] set when running from a migration to avoid infinite loop
```